ui/src/components/SearchFilters.tsx

```diff
@@ -0,0 +1,359 @@
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      region : 'All',
+      country : 'All',
+      post : 'All',
+      date : props.pickedDate,
+      timezone : 'UTC'
+    };
+
+  }
+
+  componentDidUpdate(prevProps: Readonly<IProps>, prevState: Readonly<IState>, snapshot?: any) {
+        console.log("this.state.region" + this.state.region);
+        if(this.state.region !== prevState.region) {
+           this.setState({post : 'All'});
+           this.setState({timezone : 'UTC'});
+            switch(this.state.region) {
+               case 'Euromed North' :
+                    regionPosts = euromedNorthPost;
+                    console.log("It is a euromedNorthPost.");
+                break;
+               case 'Euromed South' :
+                 regionPosts = euromedSouthPost;
+                 console.log("It is a euromedSouthPost.");
+                break;
+               case 'All' :
+                   regionPosts = allPosts;
+                   console.log("It is a allPosts.");
+                break;
+            }
+        }
+
+        if (this.state.post !== prevState.post) {
+        this.setState({country : 'All'});
+        this.setState({timezone : 'UTC'});
+            switch (this.state.post) {
+                case 'Athens' :
+                    postCountries = athensCountries;
+                    console.log("It is a athens.");
+                    break;
+                case 'Benelux':
+                    postCountries = beneluxCountries;
+                    console.log("It is a beneluxCountries.");
+                    break;
+                case 'Warsaw':
+                    postCountries = warsawCountries;
+                    console.log("It is a warsawCountries.");
+                    break;
+                case 'Berlin':
+                    postCountries = berlinCountries;
+                    console.log("It is a berlinCountries.");
+                    break;
+                case 'Paris':
+                    postCountries = parisCountries;
+                    console.log("It is a parisCountries.");
+                    break;
+                case 'Rome':
+                    postCountries = romeCountries;
+                    console.log("It is a romeCountries.");
+                    break;
+                case 'Madrid':
+                     postCountries = madridCountries;
+                    console.log("It is a madridCountries.");
+                    break;
+                case 'Albania':
+                    postCountries = albaniaCountries;
+                    console.log("It is a albaniaCountries.");
+                    break;
+                case 'All':
+                    postCountries = allCountries;
+                    console.log("It is a all countries.");
+                    break;
+                default:
+                    postCountries = allCountries;
+                    console.log("No such day exists!");
+                    break;
```

Comment:
Do you want to keep the logging? If so then just correct the last message which says 'day' but should be 'country'